File: openbis/source/java/ch/systemsx/cisd/openbis/public/resources/api/v3/as/dto/plugin/create/PluginCreation.js
define([ "stjs" ], function(stjs) {
	var PluginCreation = function() {
	};
	stjs.extend(PluginCreation, null, [], function(constructor, prototype) {
		prototype['@type'] = 'as.dto.plugin.create.PluginCreation';
		constructor.serialVersionUID = 1;
		prototype.name = null;
		prototype.description = null;
		prototype.script = null;
		prototype.pluginType = null;
		prototype.scriptType = null;
		prototype.available = true;
		prototype.entityKind = null;

		prototype.getName = function() {
			return this.name;
		};
		prototype.setName = function(name) {
			this.name = name;
		};
		prototype.getDescription = function() {
			return this.description;
		};
		prototype.setDescription = function(description) {
			this.description = description;
		};
		prototype.getScript = function() {
			return this.script;
		};
		prototype.setScript = function(script) {
			this.script = script;
		};
		prototype.getPluginType = function() {
			return this.pluginType;
		};
		prototype.setPluginType = function(pluginType) {
			this.pluginType = pluginType;
		};
		prototype.getScriptType = function() {
			return this.scriptType;
		};
		prototype.setScriptType = function(scriptType) {
			this.scriptType = scriptType;
		};
		prototype.isAvailable = function() {
			return this.available;
		};
		prototype.setAvailable = function(available) {
			this.available = available;
		};
		prototype.getEntityKind = function() {
			return this.entityKind;
		};
		prototype.setEntityKind = function(entityKind) {
			this.entityKind = entityKind;
		};
	}, {
		pluginType : "PluginType",
		scriptType : "ScriptType",
		entityKind : "EntityKind"
	});
	return PluginCreation;
})

File: openbis_api/source/java/ch/ethz/sis/openbis/generic/asapi/v3/dto/plugin/create/PluginCreation.java
/*
 * Copyright 2018 ETH Zuerich, SIS
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package ch.ethz.sis.openbis.generic.asapi.v3.dto.plugin.create;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;

import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.create.ICreation;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.create.IObjectCreation;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.entitytype.EntityKind;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.plugin.PluginType;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.plugin.ScriptType;
import ch.systemsx.cisd.base.annotation.JsonObject;

/**
 * @author Franz-Josef Elmer
 *
 */
@JsonObject("as.dto.plugin.create.PluginCreation")
public class PluginCreation implements ICreation, IObjectCreation
{
    private static final long serialVersionUID = 1L;

    @JsonProperty
    private String name;
    
    @JsonProperty
    private String description;
    
    @JsonProperty
    private PluginType pluginType;
    
    @JsonProperty
    private EntityKind entityKind;
    
    @JsonProperty
    private ScriptType scriptType;
    
    @JsonProperty
    private String script;
    
    @JsonProperty
    private boolean available = true;

    @JsonIgnore
    public String getName()
    {
        return name;
    }

    public void setName(String name)
    {
        this.name = name;
    }

    @JsonIgnore
    public String getDescription()
    {
        return description;
    }

    public void setDescription(String description)
    {
        this.description = description;
    }

    @JsonIgnore
    public PluginType getPluginType()
    {
        return pluginType;
    }

    public void setPluginType(PluginType pluginType)
    {
        this.pluginType = pluginType;
    }

    @JsonIgnore
    public EntityKind getEntityKind()
    {
        return entityKind;
    }

    public void setEntityKind(EntityKind entityKind)
    {
        this.entityKind = entityKind;
    }

    @JsonIgnore
    public ScriptType getScriptType()
    {
        return scriptType;
    }

    public void setScriptType(ScriptType scriptType)
    {
        this.scriptType = scriptType;
    }

    @JsonIgnore
    public String getScript()
    {
        return script;
    }

    public void setScript(String script)
    {
        this.script = script;
    }

    @JsonIgnore
    public boolean isAvailable()
    {
        return available;
    }

    public void setAvailable(boolean available)
    {
        this.available = available;
    }
}
